drf_spectacular/openapi.py

```diff
@@ -1016,6 +1023,47 @@ def _get_response_for_code(self, serializer, status_code, media_types=None):
             # https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#responseObject
             'description': ''
         }
+        if headers:
+            response["headers"] = headers
+        return response
+
+    def _get_response_headers_for_code(self, status_code) -> dict:
+        """
+        Extract the response header parameters.
+
+        TODO: on HTTP 201 with generics.CreateView, DRF tries to set the Location header
+        param automatically. Include that here?
+        """
+        parameters = []
+
+        # select the relevant response header parameters
+        for parameter in self.get_override_parameters():
+            if not isinstance(parameter, OpenApiParameter):
+                continue
+            if not parameter.response:
+                continue
+
+            # check if the parameter is always present, or only for this status code
+            if parameter.response is True or status_code in parameter.response:
+                parameters.append(parameter)
+
+        result = {}
+        for parameter in parameters:
+            if is_basic_type(parameter.type):
+                schema = build_basic_type(parameter.type)
+            elif is_serializer(parameter.type):
+                warn('complex types for response header parameters are not supported. '
```

Comment:
not so sure about this but ok for now i think

Reply:
I stand corrected - the `explode` and `style` attributes probably clarify this.